app/src/common/shared/com/igalia/wolvic/ui/widgets/settings/DeveloperOptionsView.java

```diff
@@ -165,10 +193,18 @@ private void setUIHardwareAcceleration(boolean value, boolean doApply) {
 
         if (doApply) {
             SettingsStore.getInstance(getContext()).setUIHardwareAccelerationEnabled(value);
-            showRestartDialog();
+            showRestartDialog(cancelUIHardwareAccelerationSelection(value));
         }
     }
 
+    private RestartDialogWidget.CancelFunction cancelUIHardwareAccelerationSelection(boolean value) {
```

Comment:
No need to define the function, you can pass a lambda to showRestartDialog(). The same for the others